src/main/scala/chisel3/stage/ChiselAnnotations.scala

```diff
@@ -61,6 +61,24 @@ case object PrintFullStackTraceAnnotation
 
 }
 
+/** On recoverable errors, this will cause Chisel to throw an exception instead of continuing.
+  */
+case object ThrowOnFirstErrorAnnotation
```

Comment:
Is this expected to be used as an annotation? If not, then this can be made `private[chisel3.stage]`.

Reply:
It will be needed by chiseltest which only exposes Annotations and not arguments, but good suggestion in general